### ISSUE TITLE
Wait and block pipeline step » different state for pipeline

### ISSUE THREAD

**Issue by jufemaiz** [buildkite/feedback#181]
Hi guys, loving Buildkite as we move a whole bunch of things from alternatives.

One thing that frustrates me a little from a UX perspective is the impact of "Wait and block pipeline" steps on the state of a pipeline.

At the moment, there's essentially four states for a step in a pipeline (as far as I can tell):

* Waiting
* Actioning
* Failed
* Passed

When a "Wait and block pipeline" step is added and the previous step(s) have completed as Passed, Buildkite indicates that the Pipeline has passed, which is invalid as it's actually in a fifth state which is not based purely on the last state of the Pipeline steps to have actioned – "Blocked" (or something similar).

This is also reflected in API feedback (Hipchat hooks, emails).

Email notification of Pipeline state - Passed (in this case [Fixed] because it was previously failing):
<img width="733" alt="screen shot 2016-08-26 at 4 46 41 pm" src="https://cloud.githubusercontent.com/assets/16481/17996514/c83178f6-6bac-11e6-980c-750a1dd055e6.png">

Hipchat notifications:
<img width="619" alt="screen shot 2016-08-26 at 4 46 51 pm" src="https://cloud.githubusercontent.com/assets/16481/17996515/c836c61c-6bac-11e6-879b-26cb114287e8.png">

It would be great to see a "Blocked" state introduced (seems to be understood by BK as the "unblocking" event is triggered and surfaced) as a feedback loop for people who can unblock the pipeline so that it may continue and truly be fixed.

Adding this so that others can contribute, based on previous conversation via Twitter @ https://twitter.com/buildkite/status/723313317012332544

**Comment by toolmantim** [comment]
Thanks for the feedback! We're working on exactly this. Stay tuned 🔜

**Comment by jufemaiz** [comment]
@toolmantim cheers mate! Thought it was good to actually use your feedback repo for this (in case others had similar requirements) :)

**Comment by gtirloni** [comment]
@jufemaiz I think that's the same thing I see on GitHub. The PR gets marked with a green checkmark but the build hasn't actually finished. When scanning the list of PRs, that could be misleading.

**Comment by jufemaiz** [comment]
@gtirloni yeah, it's not in the BK UI (it's a pause symbol) but it propagates outwards as a tick rather than maintaining an "in progress". Part of that I guess is the different use cases. @toolmantim and @lox could probably provide more insight.

**Comment by NorseGaud** [comment]
Any update on this? Block steps should not default to "passed" as this causes the PR reviewer to think it passed all tests when they review a community PR.

**Comment by toolmantim** [comment]
For people using GitHub, there’s a setting in your pipeline’s GitHub settings called “Blocked builds show in GitHub as…” which you can use to control this behavior.

**Comment by huguesb** [comment]
> For people using GitHub, there’s a setting in your pipeline’s GitHub settings called “Blocked builds show in GitHub as…” which you can use to control this behavior.

What about people not using github? Could that setting be made available more broadly via the yaml pipeline definition? Right now it seems the only way around this is to artificially keep the pipeline going with a busy loop that ties up an agent, which is really wasteful and makes input steps largely useless
